File: algorithms/276.Paint_Fence/Paint_Fence.cc
<<<<<<< HEAD
// Paint Fence
class Solution {
public:
  int numWays(int n, int k) {
    int a = 0, b = k, c = k*k;
    while (n--) {
      a = b;
      b = c;
      c = (a+b)*(k-1);
    }
    return a;
  }
=======
// Paint Fence
class Solution {
public:
  int numWays(int n, int k) {
    int a = 0, b = k, c = k*k;
    while (n--) {
      a = b;
      b = c;
      c = (a+b)*(k-1);
    }
    return a;
  }
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
};

File: algorithms/276.Paint_Fence/Paint_Fence.py
class Solution(object):
    def numWays(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: int
        """
        if not n or not k:
            return 0
        if n == 1:
            return k
        dp = [0] * (n+1)
        dp[1] = k
        dp[2] = k*(k-1) + k
        for i in range(3, n+1):
            dp[i] = (dp[i-1]+dp[i-2]) * (k - 1)
        return dp[-1]
